src/lib/validate-app-settings.js

```diff
@@ -0,0 +1,74 @@
+const joi = require('@hapi/joi');
+
+const FormsSchema = joi.object().pattern(
+  /\w{1,30}/,
+  joi.object({
+    meta: joi.object({
+      code: joi.string().required(),
+      icon: joi.string(),
+      translation_key: joi.string(),
+      subject_key: joi.string()
+    }).required(),
+    fields: joi.object().pattern(
+      /\w{1,30}/,
+      joi.object({
+        type: joi.string().required(),
+        labels: joi.object({
+          short: joi.alternatives(
+            joi.string(),
+            joi.object()
+          ),
+          tiny: joi.alternatives(
+            joi.string(),
+            joi.object()
+          )
+        }),
+        position: joi.number(),
+        flags: joi.object(),
+        length: joi.array().items(joi.number()),
+        required: joi.boolean()
```

Comment:
I've run this over standard config and it fails because some forms have fields that have `validations` and `range` properties. 
I'm not sure if these fields are even used, but we should definitely make sure. 

I think we should run this against a number of SMS projects configurations and see if they pass.